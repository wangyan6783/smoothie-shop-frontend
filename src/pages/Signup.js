import React, { useState } from "react";
import { Link, withRouter } from "react-router-dom";
import { connect } from "react-redux";
import { setCurrentUser } from "../redux/user/user.actions";

function Signup(props) {
  const [username, setUsername] = useState("");
  const [password, setPassword] = useState("");
  const [signupError, setSignupError] = useState(false);

  function handleUsername(e) {
    setUsername(e.target.value);
  }

  function handlePassword(e) {
    setPassword(e.target.value);
  }

  function handleSubmit(e) {
    e.preventDefault();
    setUsername("");
    setPassword("");
    fetch("http://localhost:5000/users/signup", {
      method: "POST",
      headers: {
        Accept: "application/json",
        "Content-Type": "application/json"
      },
      body: JSON.stringify({
        username,
        password
      })
    })
      .then(r => r.json())
      .then(data => {
        console.log(data);
        if (data === "error") {
          setSignupError(true);
        } else {
          props.setCurrentUser({ username: data.username });
          props.history.push("/");
        }
      })
      .catch(err => console.log(err));
  }

  return (
    <div className="signup">
      <form className="form" onSubmit={handleSubmit}>
        <h1 className="heading-1 mb-md">Sign up</h1>
        {signupError ? <div>Username already taken</div> : null}
        <div>
          <label className="form-text" htmlFor="username">
            Username
          </label>
          <input
            className="form-input"
            type="text"
            name="username"
            value={username}
            onChange={handleUsername}
          ></input>
        </div>
        <div className="mb-sm">
          <label className="form-text" htmlFor="password">
            Password
          </label>
          <input
            className="form-input mb-sm"
            type="password"
            name="password"
            value={password}
            onChange={handlePassword}
          ></input>
        </div>
        <button className="btn form-btn mb-sm" type="submit">
          Sign up
        </button>
        <p className="form-text">
          {`Already have an account? `}
          <Link className="form-link" to="/login">
            Log In
          </Link>
        </p>
      </form>
    </div>
  );
}

const mapDispatchToProps = dispatch => ({
  setCurrentUser: user => dispatch(setCurrentUser(user))
});

export default withRouter(connect(null, mapDispatchToProps)(Signup));
