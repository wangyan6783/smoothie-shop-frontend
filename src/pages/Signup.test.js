import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";
import Signup from "./Signup";

describe("Signup Component", () => {
  it("render correctly", () => {
    const signupComponent = renderer
      .create(
        <Router>
          <Signup />
        </Router>
      )
      .toJSON();
    expect(signupComponent).toMatchSnapshot();
  });
});
