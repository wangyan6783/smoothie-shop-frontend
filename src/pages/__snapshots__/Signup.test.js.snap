// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup Component render correctly 1`] = `
<div
  className="signup"
>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <h1
      className="heading-1 mb-md"
    >
      Sign up
    </h1>
    <div>
      <label
        className="form-text"
        htmlFor="username"
      >
        Username
      </label>
      <input
        className="form-input"
        name="username"
        onChange={[Function]}
        type="text"
      />
    </div>
    <div
      className="mb-sm"
    >
      <label
        className="form-text"
        htmlFor="password"
      >
        Password
      </label>
      <input
        className="form-input mb-sm"
        name="password"
        onChange={[Function]}
        type="text"
      />
    </div>
    <button
      className="btn form-btn mb-sm"
      type="submit"
    >
      Sign up
    </button>
    <p
      className="form-text"
    >
      Already have an account?
       
      <a
        className="form-link"
        href="/login"
        onClick={[Function]}
      >
        Log In
      </a>
    </p>
  </form>
</div>
`;
